shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -220,57 +197,513 @@ - (CGRect)accessibilityFrame {
+}
+
+- (UITextPosition*)closestPositionToPoint:(CGPoint)point withinRange:(UITextRange*)range {
+  return nil;
+}
+
+- (NSArray*)selectionRectsForRange:(UITextRange*)range {
+  return @[];
+}
+
+- (UITextRange*)characterRangeAtPoint:(CGPoint)point {
+  return nil;
+}
+
+- (void)insertText:(NSString*)text {
+}
+
+- (void)deleteBackward {
+}
+
+@end
+
+@implementation TextInputSemanticsObject {
+  InactiveTextInput* _inactive_text_input;
+}
+
+- (instancetype)initWithBridge:(shell::AccessibilityBridge*)bridge uid:(int32_t)uid {
+  self = [super initWithBridge:bridge uid:uid];
+
+  if (self) {
+    _inactive_text_input = [[InactiveTextInput alloc] init];
+  }
+
+  return self;
+}
+
+- (void)dealloc {
+  [_inactive_text_input release];
+  [super dealloc];
+}
+
+#pragma mark - UIAccessibility overrides
+
+- (void)setSemanticsNode:(const blink::SemanticsNode*)node {
+  [super setSemanticsNode:node];
+  _inactive_text_input.text = @(node->value.data());
+  if ([self node].HasFlag(blink::SemanticsFlags::kIsFocused)) {
+    // The text input view must have a non-trivial size for the accessibility
+    // system to send text editing events.
+    [self bridge]->textInputView().frame = CGRectMake(0.0, 0.0, 1.0, 1.0);
+  }
+}
+
+/**
+ * The UITextInput whose accessibility properties we present to UIKit as
+ * substitutes for Flutter's text field properties.
+ *
+ * When the field is currently focused (i.e. it is being edited), we use
+ * the FlutterTextInputView used by FlutterTextInputPlugin. Otherwise,
+ * we use an InactiveTextInput.
+ */
+- (UIView<UITextInput>*)textInputSurrogate {
+  if ([self node].HasFlag(blink::SemanticsFlags::kIsFocused)) {
+    return [self bridge]->textInputView();
+  } else {
+    return _inactive_text_input;
+  }
+}
+
+- (UIView*)textInputView {
+  return [self textInputSurrogate];
+}
+
+- (void)accessibilityElementDidBecomeFocused {
+  [[self textInputSurrogate] accessibilityElementDidBecomeFocused];
+}
+
+- (void)accessibilityElementDidLoseFocus {
+  [[self textInputSurrogate] accessibilityElementDidLoseFocus];
+}
+
+- (BOOL)accessibilityElementIsFocused {
+  return [self node].HasFlag(blink::SemanticsFlags::kIsFocused);
+}
+
+- (BOOL)accessibilityActivate {
+  return [[self textInputSurrogate] accessibilityActivate];
+}
+
+- (NSString*)accessibilityLabel {
+  return [self textInputSurrogate].accessibilityLabel;
+}
+
+- (NSString*)accessibilityHint {
+  return [self textInputSurrogate].accessibilityHint;
+}
+
+- (NSString*)accessibilityValue {
+  return [self textInputSurrogate].accessibilityValue;
+}
+
+- (UIAccessibilityTraits)accessibilityTraits {
+  // Adding UIAccessibilityTraitKeyboardKey to the trait list so that iOS treat it like
+  // a keyboard entry control, thus adding support for text editing features, such as
+  // pinch to select text, and up/down fling to move cursor.
+  return [self textInputSurrogate].accessibilityTraits | UIAccessibilityTraitKeyboardKey;
+}
+
+- (CGPoint)accessibilityActivationPoint {
```

Comment:
Why do we need to move the activation point to the center? Isn't that the default?

Reply:
Done.